components/src/core/DrawerLayout/DrawerLayout.tsx

```diff
@@ -1,10 +1,20 @@
-import React, { ReactNode } from 'react';
-import { makeStyles } from '@material-ui/core/styles';
+import React from 'react';
+import { makeStyles, useTheme } from '@material-ui/core/styles';
 import PropTypes from 'prop-types';
+import { DrawerProps, useMediaQuery } from '@material-ui/core';
+import { Drawer } from '../Drawer';
 
 export type DrawerLayoutProps = {
+    // Page's body
     children: React.ReactNode;
-    drawer: ReactNode;
+
+    // Drawer component to be embedded
+    drawer: JSX.Element;
```

Comment:
I'll echo Evan's comment that we need to to be able to do this by directly passing in a drawer, or through a FC